src/cargo/sources/git/utils.rs

```diff
@@ -296,52 +294,61 @@ impl<'a> GitCheckout<'a> {
             info!("update submodules for: {:?}", repo.workdir().unwrap());
 
             for mut child in repo.submodules()?.into_iter() {
-                child.init(false)?;
-                let url = child.url().chain_error(|| {
-                    internal("non-utf8 url for submodule")
+                update_submodule(repo, &mut child, cargo_config).chain_error(|| {
+                    human(format!("Failed to update submodule `{}`",
```

Comment:
After very small sampling I understood that `human` variants start with uppercase but `internal` with lowercase but I will fix this soonish.

I made this a `human` variant as it needs to be shown without `--verbose` without checking any of the reporting code.